### PR TITLE
EZP-30314: Content is cached even after deletion

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30314](https://jira.ez.no/browse/EZP-30314)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.4`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/TrashHandler.php [hunk 1]
```diff
@@ -134,15 +138,19 @@ public function deleteTrashItem($trashedId)
...
 
-        if (!empty($tags)) {
-            $this->cache->invalidateTags($tags);
-        }
+        $this->cache->invalidateTags([
+            'content-' . $return->contentId,
+            'content-fields-' . $return->contentId,
+            'location-' . $trashedId,
+            'location-path-' . $trashedId,
+        ] + $tags);
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Cache/TrashHandler.php]: This is not going to work, see what happens if:
```php
var_dump(['content-fields-1'] + ['content-fields-2']);
```
Additionally, if tests won't crash here, you'll need to add missing coverage :upside_down_face:

Comment (ViniTou) [reply eZ/Publish/Core/Persistence/Cache/TrashHandler.php]: Yup, you are right, I have fixed it also for trashSubtree now.

**Comment by andrerom** [comment]
Is there anyway we can add coverage in integration tests ?

**Comment by ViniTou** [comment]
@alongosz 
@andrerom 
@adamwojs 

I have added some simple checkup for existing integration test, could you take a look on that?

File: eZ/Publish/API/Repository/Tests/TrashServiceTest.php [hunk 2]
```diff
@@ -734,21 +734,25 @@ public function testEmptyTrash()
...
         /* BEGIN: Use Case */
-        $this->createTrashItem();
+        $trashItem = $this->createTrashItem();
 
         // Empty the trash
         $trashService->emptyTrash();
 
+        // Try to load content
+        $this->expectException(NotFoundException::class);
+        $contentService->loadContent($trashItem->contentId);
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/TrashServiceTest.php]: does the test of the test execute now?

File: eZ/Publish/API/Repository/Tests/TrashServiceTest.php [hunk 3]
```diff
@@ -783,6 +788,10 @@ public function testDeleteTrashItem()
         // Empty the trash
         $trashService->deleteTrashItem($trashItem);
 
+        // Try to load Content
+        $this->expectException(NotFoundException::class);
+        $contentService->loadContent($trashItem->contentId);
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/TrashServiceTest.php]: same question here

File: eZ/Publish/API/Repository/Tests/TrashServiceTest.php (hunk 2 above)
Comment (ViniTou) [reply eZ/Publish/API/Repository/Tests/TrashServiceTest.php]: Now yes, good catch, thanks!

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**